test/wallet-http-test.js

```diff
@@ -0,0 +1,558 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const {rimraf, sleep} = require('./util/common');
+
+const {
+  initFullNode,
+  initSPVNode,
+  initNodeClient,
+  initWalletClient,
+  initWallet,
+  generateInitialBlocks,
+  generateTxs
+} = require('./util/regtest');
+
+const testPrefix = '/tmp/bcoin-fullnode';
+const spvTestPrefix = '/tmp/bcoin-spvnode';
+const genesisTime = 1534965859;
+const genesisDate = new Date(genesisTime * 1000);
+
+const ports = {
+  full: {
+    p2p: 49331,
+    node: 49332,
+    wallet: 49333
+  },
+  spv: {
+    p2p: 49431,
+    node: 49432,
+    wallet: 49433
+  }
+};
+
+describe('Wallet TX HTTP Pagination', function() {
+  this.timeout(60000);
+
+  let node, spvnode, wallet = null;
+  let nclient, wclient, spvwclient = null;
+  let coinbase = null;
+  let unconfirmedTime = null;
+  let unconfirmedDate = null;
+
+  before(async () => {
+    await rimraf(testPrefix);
+    await rimraf(spvTestPrefix);
+
+    node = await initFullNode({
+      ports,
+      prefix: testPrefix,
+      logLevel: 'none'
+    });
+
+    spvnode = await initSPVNode({
+      ports,
+      prefix: spvTestPrefix,
+      logLevel: 'none'
+    });
+
+    nclient = await initNodeClient({ports: ports.full});
+    wclient = await initWalletClient({ports: ports.full});
+    spvwclient = await initWalletClient({ports: ports.spv});
+    wallet = await initWallet(wclient);
+    await initWallet(spvwclient);
+
+    await wclient.execute('selectwallet', ['test']);
+    coinbase = await wclient.execute('getnewaddress', ['blue']);
+
+    await spvwclient.execute('selectwallet', ['test']);
+
+    await generateInitialBlocks({
+      nclient,
+      wclient,
+      spvwclient,
+      coinbase,
+      genesisTime,
+      blocks: 125
+    });
+
+    // Generate unconfirmed transactions for the
+    // fullnode and spv wallet
+    unconfirmedTime = Math.floor(Date.now() / 1000);
+    unconfirmedDate = new Date(unconfirmedTime * 1000);
+
+    // TODO remove this
+    await sleep(5000);
+
+    await generateTxs({
+      wclient,
+      spvwclient,
+      count: 19,
+      amount: 0.0001,
+      gap: 6,
+      sleep: 2000
+    });
+
+    // TODO remove this. It's necessary to wait because
+    // the transactions need to be broadcast to the
+    // spv node. However the above await continues
+    // before broadcast.
+    await sleep(5000);
+  });
+
+  after(async () => {
+    await wallet.close();
+    await wclient.close();
+    await spvwclient.close();
+    await nclient.close();
+    await node.close();
+    await spvnode.close();
+  });
+
+  describe('full node', function() {
+    describe('confirmed and unconfirmed txs (dsc)', function() {
+      it('first page', async () => {
+        const history = await wclient.get('/wallet/test/tx/history', {
```

Comment:
I think it would be ideal to include updates to `bclient` as well for wrappers around the HTTP methods themselves. Since getting that merged in sync with getting this merged in sync may be challenging, this is ok for now

Reply:
Preferably, I think many tests would not depend on bclient, as it's now a separate package. I minimal HTTP client could be used instead. For the reasons you mention, and for separation of testing.